File: core/src/nl/t64/game/rpg/screens/inventory/InventoryUtils.java
package nl.t64.game.rpg.screens.inventory;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.scenes.scene2d.ui.Table;
import com.badlogic.gdx.scenes.scene2d.ui.Window;
import lombok.Getter;
import nl.t64.game.rpg.Utils;
import nl.t64.game.rpg.components.party.HeroItem;
import nl.t64.game.rpg.components.party.PartyContainer;
import nl.t64.game.rpg.constants.Constant;
import nl.t64.game.rpg.screens.ScreenUI;


public final class InventoryUtils {

    @Getter
    private static HeroItem selectedHero = null;

    private InventoryUtils() {
        throw new IllegalCallerException("InventoryUtils class");
    }

    public static String getSelectedHeroId() {
        PartyContainer party = Utils.getGameData().getParty();
        if (!party.containsExactlyEqualTo(selectedHero)) {
            selectedHero = party.getHero(0);
        }
        return selectedHero.getId();
    }

    public static void selectPreviousHero() {
        selectedHero = Utils.getGameData().getParty().getPreviousHero(selectedHero);
    }

    public static void selectNextHero() {
        selectedHero = Utils.getGameData().getParty().getNextHero(selectedHero);
    }

    public static void setWindowDeselected(Table container) {
        Window parent = (Window) container.getParent();
        parent.getTitleLabel().getStyle().fontColor = Color.BLACK;
    }

    public static void setWindowSelected(Table container) {
        Window parent = (Window) container.getParent();
        parent.getTitleLabel().getStyle().fontColor = Constant.DARK_RED;
    }

    public static ScreenUI getScreenUI() {
        return Utils.getScreenManager().getCurrentScreenToLoad().getScreenUI();
    }

}

File: core/src/nl/t64/game/rpg/screens/inventory/InventoryUtils.kt
package nl.t64.game.rpg.screens.inventory

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.scenes.scene2d.ui.Table
import com.badlogic.gdx.scenes.scene2d.ui.Window
import nl.t64.game.rpg.Utils.gameData
import nl.t64.game.rpg.Utils.screenManager
import nl.t64.game.rpg.components.party.HeroItem
import nl.t64.game.rpg.constants.Constant
import nl.t64.game.rpg.screens.ScreenUI


object InventoryUtils {

    private var selectedHero: HeroItem? = null

    fun getSelectedHero(): HeroItem {
        return selectedHero!!
    }

    fun getSelectedHeroId(): String {
        val party = gameData.party
        if (selectedHero == null || !party.containsExactlyEqualTo(selectedHero!!)) {
            selectedHero = party.getHero(0)
        }
        return selectedHero!!.id
    }

    fun selectPreviousHero() {
        selectedHero = gameData.party.getPreviousHero(selectedHero!!)
    }

    fun selectNextHero() {
        selectedHero = gameData.party.getNextHero(selectedHero!!)
    }

    fun setWindowDeselected(container: Table) {
        val parent = container.parent as Window
        parent.titleLabel.style.fontColor = Color.BLACK
    }

    fun setWindowSelected(container: Table) {
        val parent = container.parent as Window
        parent.titleLabel.style.fontColor = Constant.DARK_RED
    }

    fun getScreenUI(): ScreenUI = screenManager.getCurrentParchmentScreen().getScreenUI()

}
